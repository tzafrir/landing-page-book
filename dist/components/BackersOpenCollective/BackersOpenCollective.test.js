'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _enzyme = require('enzyme');

var _enzymeAdapterReact = require('enzyme-adapter-react-16');

var _enzymeAdapterReact2 = _interopRequireDefault(_enzymeAdapterReact);

var _BackersOpenCollective = require('./BackersOpenCollective');

var _BackersOpenCollective2 = _interopRequireDefault(_BackersOpenCollective);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

(0, _enzyme.configure)({ adapter: new _enzymeAdapterReact2.default() });

describe('Backers', function () {
  var props = void 0;
  var mounted = void 0;
  var backersOpenCollective = function backersOpenCollective() {
    if (!mounted) {
      mounted = (0, _enzyme.shallow)(_react2.default.createElement(_BackersOpenCollective2.default, props));
    }
    return mounted;
  };
  describe('BackersOpenCollective', function () {
    beforeEach(function () {
      props = {
        collective: 'webpack'
      };
      mounted = undefined;
    });

    beforeEach(function () {
      props = {
        collective: 'chrome'
      };
      mounted = undefined;
    });

    it('should render', function () {
      expect((0, _enzyme.shallow)(_react2.default.createElement(_BackersOpenCollective2.default, props))).toMatchSnapshot();
    });

    it('always renders a div', function () {
      var divs = backersOpenCollective().find('div');

      expect(divs.length).toBeGreaterThan(0);
    });

    it('sort function', function () {
      var backers = [{ contributionAmount: 1, name: 'Nick', image: 'image.jpg' }, { contributionAmount: 23, name: 'John', image: 'image.jpg' }];
      backersOpenCollective().instance().sort(backers, 'asc');

      expect(backersOpenCollective().state('backersOrdered')[0].name).toEqual('Nick');
    });
  });
});