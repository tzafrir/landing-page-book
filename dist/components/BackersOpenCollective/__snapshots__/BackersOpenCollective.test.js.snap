// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backers BackersOpenCollective should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BackersOpenCollective
    collective="chrome"
    imageBordered={true}
    imageGray={false}
    imageRounded={true}
    loadingClass="fa fa-circle-o-notch fa-spin"
    loadingText="Loading..."
    sortDirection="desc"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="text-center"
        >
          <Icon
            className=""
            icon="fa fa-circle-o-notch fa-spin"
            onClick={[Function]}
            style={Object {}}
            title=""
          />
          <p>
            Loading...
          </p>
        </div>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Icon
              className=""
              icon="fa fa-circle-o-notch fa-spin"
              onClick={[Function]}
              style={Object {}}
              title=""
            />,
            <p>
              Loading...
            </p>,
          ],
          "className": "text-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "icon": "fa fa-circle-o-notch fa-spin",
              "onClick": [Function],
              "style": Object {},
              "title": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Loading...",
            },
            "ref": null,
            "rendered": "Loading...",
            "type": "p",
          },
        ],
        "type": "div",
      },
      false,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="text-center"
          >
            <Icon
              className=""
              icon="fa fa-circle-o-notch fa-spin"
              onClick={[Function]}
              style={Object {}}
              title=""
            />
            <p>
              Loading...
            </p>
          </div>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Icon
                className=""
                icon="fa fa-circle-o-notch fa-spin"
                onClick={[Function]}
                style={Object {}}
                title=""
              />,
              <p>
                Loading...
              </p>,
            ],
            "className": "text-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "icon": "fa fa-circle-o-notch fa-spin",
                "onClick": [Function],
                "style": Object {},
                "title": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Loading...",
              },
              "ref": null,
              "rendered": "Loading...",
              "type": "p",
            },
          ],
          "type": "div",
        },
        false,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
